### PR TITLE
Move logout/login to non-fatal test

### PR DESCRIPTION
**Pull request by dzedro**
I would like to hear opinion on line 23, I guess DM_NEEDS_USERNAME is behavior from past, but today we have lot of DMs on many versions, checking for user prompt is more convenient than setting DM_NEEDS_USERNAME on every test

[update_13.2](http://10.100.12.155/tests/4634#step/first_boot/9)

### CODE REVIEW

File: tests/installation/first_boot.pm [hunk 1]
```diff
@@ -20,7 +20,7 @@ use testapi;
 sub handle_login {
     mouse_hide();
     wait_still_screen;
-    if (get_var('DM_NEEDS_USERNAME')) {
+    if (get_var('DM_NEEDS_USERNAME') || check_screen 'displaymanager-username-prompt', 3) {
```
Comment (okurz) [inline tests/installation/first_boot.pm]: would `check_screen '…', 0` also work here? There is a `wait_still_screen` in before so there should not be any change

File: tests/installation/first_boot.pm [hunk 2]
```diff
@@ -93,6 +93,7 @@ sub run() {
         }
         assert_and_click 'logout-button';                         # press logout
     }
+    assert_screen 'displaymanager';
```
Comment (okurz) [inline tests/installation/first_boot.pm]: that one is fine :-)

File: tests/installation/first_boot.pm (hunk 1 above)
Comment (dzedro) [inline tests/installation/first_boot.pm]: should work, but we will be strict so no check screen :)

**Comment by dzedro** [comment]
I'm fine with using DM_NEEDS_USERNAME but it can get tricky
e.g. if I comapre [update_staging](https://openqa.opensuse.org/tests/333522#step/first_boot/7) and [update_13.2](https://openqa.opensuse.org/tests/333194#step/first_boot/7)
update_staging doesn't need DM_NEEDS_USERNAME but update_13.2 does need DM_NEEDS_USERNAME
Maybe update_staging will need DM_NEEDS_USERNAME some day or maybe not, we will see.

**Comment by dzedro** [comment]
And another thing I was thinking about, move this logout/login into separate not fatal test, I think it is good place for it, but because DM is always changing/failing and first_boot should be **stable** as it is milestone, then this code should move and leave it stable :)

**Comment by okurz** [comment]
yes, very good point. `first_boot` should concentrate on booting, i.e. as soon as either the auto-login desktop shows or the display manager with login screen it should pass onto the next module

**Comment by DimStar77** [comment]
DM_NEEDS_USERNAME as variable is very handy - in the other_test framework for example we 'switch' DM for example during setup - so on first login we might have xdm (needing username to be typed), then on next reboot we end up with lightdm; the test reconfigures DM_NEEDS_USERNAME in this case for example

**Comment by dzedro** [comment]
Login moved from first_boot

[TW xfce](http://10.100.12.155/tests/4656#step/user_gui_login/2)
[TW lxde](http://10.100.12.155/tests/4657#step/user_gui_login/3)
[TW gnome](http://10.100.12.155/tests/4655#step/user_gui_login/3)

File: tests/x11/user_gui_login.pm [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        my $command = check_var('DESKTOP', 'gnome') ? 'gnome-session-quit' : 'lxsession-logout';
+        x11_start_program("$command");    # opens logout dialog
+        assert_screen 'logoutdialog' unless check_var('DESKTOP', 'gnome');
+    }
+    else {
+        my $key = check_var('DESKTOP', 'xfce') ? 'alt-f4' : 'ctrl-alt-delete';
+        send_key_until_needlematch 'logoutdialog', "$key";    # opens logout dialog
+    }
+    assert_and_click 'logout-button';                         # press logout
+    assert_screen 'displaymanager';                           # wait for DM, then try to login
```
Comment (DimStar77) [inline tests/x11/user_gui_login.pm]: I'd expect the assert_screen displaymanager to be part of handle_login

Comment (okurz) [inline tests/x11/user_gui_login.pm]: It can be seen as both, a pre condition as well as a post condition so it does not really matter

Comment (dzedro) [inline tests/x11/user_gui_login.pm]: moved into handle_login

**Comment by okurz** [comment]
looks quite good now, let's try it out

**Pull request merged**